### ISSUE TITLE
naming issues: dotnet-sdk dotnet-runtime dotnet-aspnetcore

### ISSUE THREAD

**Issue by Plarpoon** [NixOS/nixpkgs#260915]
- Package name: dotnet-sdk - dotnet-runtime - dotnet-aspnetcore
- Latest released version: 7.0
- Current version on the unstable channel: 6.0 LTS
- Current version on the stable/release channel: 6.0 LTS
- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**
@kuznero @mdarocha 

-----

Hi all, this is my first issue in here so I hope I am doing everything alright.

I just noticed that there is a bit of confusion with the naming and packaging of .NET and I would have liked to discuss with you if there was a possibility of changing it to the same naming and versioning scheme that [Microsoft uses](https://dotnet.microsoft.com/en-us/download).

I have noticed that at the moment we have a dotnet-runtime and dotnet-sdk packages that are running the 6.0 version, which is the LTS. I would like to call that one LTS.

We then have dotnet-runtime_7 and dotnet-sdk_7 which are actually the stable releases and shouldn't require to be mentioned the version 7 to be installed.

We then have dotnet-runtime_8 and dotnet-sdk_8 which except for the version number being different have the same exact schema as for the stable version but are in reality the preview version of version 8.

Same exact thing happening for the aspnetcore packages

I suggest we do as follows:

rename:
**dotnet-runtime** => **dotnet-runtime-lts**
**dotnet-sdk** => **dotnet-sdk-lts**
**dotnet-aspnetcore** => **dotnet-aspnetcore-lts**

**dotnet-runtime_7** => **dotnet-runtime**
**dotnet-sdk_7** => **dotnet-sdk**
**dotnet-aspnetcore_7** => **dotnet-aspnetcore**

**dotnet-runtime_8** => **dotnet-runtime-preview**
**dotnet-sdk_8** => **dotnet-sdk-preview**
**dotnet-aspnetcore_8** => **dotnet-aspnetcore-preview**

I am obviously open to critique in case you think this is not a good idea but at the moment I genuinely believe this is the healthiest thing to do for several reasons.

1) We would keep naming parity with Microsoft making it clear at first sight what are we looking for when searching for packages. This is especially nice for very new users like me that are just discovering Nix for the first time 😃 

2) The numbers are just going to increase and sooner or later we will have a big amount of version in our packages without a proper schema. With this change we would be able to always pin in our configuration files to always install either from the stable, preview or lts channel. If for some reason we need an old unsupported version we can always just install them by selection "**dotnet-sdk_6**" like we have now!

I believe this would definitely improve the current situation, if I can be of any help please let me know! Ping me at will and we can chat here or any other social networks, I am almost always available!
-----

Note for maintainers: Please tag this issue in your PR.

**Comment by Plarpoon** [comment]
While still thinking that my original proposal is better so in one go we are done with all changes I don't dislike your proposal at all, as that would still be a big improvement over the current situation.

**Comment by eclairevoyant** [comment]
I originally +1'd this based on some assumptions that turned out to be wrong. After double checking the MS site, I don't think the original progosal is even "Microsoft naming convention" but rather a bit made up.

As per [this page](https://dotnet.microsoft.com/en-us/download/dotnet), they do not have this -LTS or -Preview distinction; instead there is a list of versions, their support type (STS vs LTS) and the support phase. However, [as of next month](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#cadence), we will have _two LTS versions in active support_ (6.0 and 8.0), and an STS version (7.0) and calling 8.0 "preview" won't really make sense as it's both the current and the LTS.

I would be ok with _warning_ users when they use the alias with no version, but not this change, which I believe will add more confusion over the coming years.

**Comment by Plarpoon** [comment]
If you look at the link you have yourself provided you can see with what I have confused myself, it used to be called "current releases" and now it is indeed called "sts". While preview is just what they call as umbrella term every phase before release. Thank you for correcting and updating me on that one!

But yes, having two active LTS versions would indeed be problematic, on that I fully agree with you, odd that they are doing that but that is besides the point.

I still think there are two different issues we should tackle on the dotnet releases.

1) Some packages have no version and some do. Which is inconsistent at least within nixpkgs itself.

2) There is no reliable way to always target the most recent stable version of dotnet without explicitly declaring the specific target version.


If you want to propose something that would tackle both issues I can even update the first post with your idea @eclairevoyant! I am trying to make this thing better so your ideas are as good as mine! 😃

**Comment by eclairevoyant** [comment]
Well thankfully there will only be one STS active at any point, so I think adding -sts is fine. LTS might be confusing, but -lts to point to the most-recent LTS, I would be ok with.
And I suggest we not add -preview as that only applies to a short time frame prior to the release.

TLDR I'm ok with `-sts` and `-lts`, but I wouldn't add `-preview`

**Comment by Plarpoon** [comment]
I like your proposal, if anyone else agrees and has nothing against this change please write it as a comment and I will change the first post!

**Comment by eclairevoyant** [comment]
The latter is not a thing in nixpkgs except in one package I could find (netcoredbg). For a second I also thought BeatSaberModManager had the issue, but quickly saw it's not even used in the nix expression.

**Comment by Plarpoon** [comment]
I will be very happy if this change gets applied, at least we can streamline the process quite a bit

**Comment by Plarpoon** [comment]
I think we are all in agreement @mdarocha, in such a spirit I have edited the first post to reflect exactly that change!

**Comment by Plarpoon** [comment]
I have never opened a PR before in my life @mdarocha so if you see that I landed flat on my face can you please tell me?

https://github.com/NixOS/nixpkgs/pull/261695

**Comment by l3m** [comment]
> So guess we can add the `-lts` and `-sts` aliases without much trouble - is everybody ok with that?

The -sts suffix is surprising. Because of how the dotnet release schedule works, it is only ever useful every other year. As new major versions are released at the end of the year, I will count them for the following year in this table. 

| Year | Latest | LTS | STS |  
| :----:|:------:|:---:|:----:|
| 2021 | 5.0 | 3.1 | 5.0 | 
| 2022 | 6.0 | 6.0 | not meaningful |
| 2023 | 7.0 | 6.0 | 7.0 |
| 2024 | 8.0 | 8.0 | not meaningful |

Not meaningful meaning that the last non-LTS version is out of support and generally one is advised to use the current LTS, not an out-of-support version.

Therefore, while -lts makes sense, I think it's better to either have no suffix for the latest dotnet version (as it might be lts or not, depending on the year), or use -current, -latest or something like that.

**Comment by YoshiRulz** [comment]
Who is this for? If you want max. stability, pin Nixpkgs, or at least be explicit about the version. If you're after the latest release (with `nix-shell` for example), you'd expect the unsuffixed attribute to be that.
